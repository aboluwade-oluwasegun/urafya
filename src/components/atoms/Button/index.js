import React, { Component } from "react";
import PropTypes from "prop-types";

class Button extends Component {
  static propTypes = {
    onClick: PropTypes.func.isRequired,
    title: PropTypes.string.isRequired,
    icon: PropTypes.string,
    showIcon: PropTypes.bool,
  };

  static defaultProps = {
    showIcon: false,
  };

  render() {
    const { icon, showIcon, onClick, title } = this.props;
    return (
      <div onClick={onClick} className="mybtncurve btn btn-primary shadow-sm">
        {showIcon && <i className={icon}></i>}
        {title}
      </div>
    );
  }
}

export default Button;
