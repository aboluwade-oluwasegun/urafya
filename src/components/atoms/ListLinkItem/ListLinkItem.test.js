import React from "react";
import { shallow } from "enzyme";
import Button from ".";
import LinkListItem from ".";

describe("Button component", () => {
  const shallowRender = (props = {}) =>
    shallow(
      <LinkListItem
        link="/admin/users"
        linkText="Users"
        icon="fa fa-user"
        {...props}
      />
    );
  it("renders correctly", () => {
    expect(shallowRender()).toMatchSnapshot();
  });

  it("renders the icon", () => {
    const wrapper = shallowRender();
    expect(wrapper.find("i").exists()).toBe(true);
  });

  it("renders the Link component", () => {
    const wrapper = shallowRender();
    expect(wrapper.find("Link").exists()).toBe(true);
  });

  it("renders the correct tile for the Link component", () => {
    const linkText = "New User Button";
    const wrapper = shallowRender({ linkText });
    expect(wrapper.find("span").exists()).toBe(true);
    expect(wrapper.find("span").props().children).toContain(linkText);
  });

  it("has the correct link", () => {
    const link = "/admin/users";
    const wrapper = shallowRender({ link });
    expect(wrapper.find("Link").props().to).toBe(link);
  });
});
