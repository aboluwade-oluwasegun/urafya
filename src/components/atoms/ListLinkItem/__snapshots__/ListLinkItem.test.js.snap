// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LinkListItem
    icon="fa fa-user"
    link="/admin/users"
    linkText="Users"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        className="nav-link"
        replace={false}
        to="/admin/users"
>
        <i
                className="fa fa-user"
        />
        <span>
                Users
        </span>
</Link>,
      "className": "nav-item",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <i
            className="fa fa-user"
/>,
          <span>
            Users
</span>,
        ],
        "className": "nav-link",
        "replace": false,
        "to": "/admin/users",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fa fa-user",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Users",
          },
          "ref": null,
          "rendered": "Users",
          "type": "span",
        },
      ],
      "type": [Function],
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          className="nav-link"
          replace={false}
          to="/admin/users"
>
          <i
                    className="fa fa-user"
          />
          <span>
                    Users
          </span>
</Link>,
        "className": "nav-item",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <i
              className="fa fa-user"
/>,
            <span>
              Users
</span>,
          ],
          "className": "nav-link",
          "replace": false,
          "to": "/admin/users",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa fa-user",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Users",
            },
            "ref": null,
            "rendered": "Users",
            "type": "span",
          },
        ],
        "type": [Function],
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
